test/support/data_case.ex

```diff
@@ -0,0 +1,49 @@
+defmodule SingForNeedsTest.DataCase do
+    @moduledoc """
+    This module defines the setup for tests requiring
+    access to the application's data layer.
+     You may define functions here to be used as helpers in
+    your tests.
+     Finally, if the test case interacts with the database,
+    it cannot be async. For this reason, every test runs
+    inside a transaction which is reset at the beginning
+    of the test unless the test case is marked as async.
+    """
+
+    use ExUnit.CaseTemplate
+
+    using do
+      quote do
+        alias SingForNeedsTest.Repo
+
+        import Ecto
+        import Ecto.Changeset
+        import Ecto.Query
+        import SingForNeedsTest.DataCase
```

Comment:
Forgive me, it's been a while since I've looked at case templates and there may be an obvious answer. Why does it need to import itself?

Reply:
@SherSpock It's autogenerated. A convention. Blame @chrismccord !! 😁